haystack/schema.py

```diff
@@ -100,11 +103,37 @@ def __init__(
         if id:
             self.id: str = str(id)
         else:
-            self.id: str = self._get_id(id_hash_keys)
+            self.id: str = self._get_id(id_hash_keys=id_hash_keys, id_hash_from=id_hash_from)
 
-    def _get_id(self, id_hash_keys):
-        final_hash_key = ":".join(id_hash_keys) if id_hash_keys else str(self.content)
-        return '{:02x}'.format(mmh3.hash128(final_hash_key, signed=False))
+
+    def _get_id(self, 
+        id_hash_keys: Optional[List[str]] = None,
+        id_hash_from:  Optional[List[Literal["content", "meta"]]] = None
+    ):
+        """
+        Generate the id of a document by creating the hash of strings. By default the content of a document is 
+        used to generate the hash. There are two ways of modifying the generated id of a document. Either static keys
+        or a selection of the content. 
+        :param id_hash_keys: Optional list of strings that are used to generate the hash. 
+        :param id_hash_from: Optional list of fields that should be dynamically used to generate the hash. 
+        """
+
+        if id_hash_keys is None and id_hash_from is None: 
+            return '{:02x}'.format(mmh3.hash128(str(self.content), signed=False)) 
+
+        final_hash_key = "" 
+        if id_hash_keys is not None:
+            final_hash_key += ":".join(id_hash_keys)
+
+        if id_hash_from is not None:
+            if "content" in id_hash_from: 
```

Comment:
Let's make this dynamic and allow accessing any attribute of `Document` (e.g. `Document.score` or `Document.content_type`)